### PR TITLE
Work around spurious timer notifications by re-arming timer

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: lib/core/monitor_impl.toit [hunk 1]
```diff
@@ -49,7 +49,7 @@ class __Monitor__:
     if deadline:
       timer = self.acquire_timer_
       // Arrange for the notify_ method to be called if the timeout expires.
-      timer.arm this deadline
+      timer.set_target this
```
Comment (erikcorry) [inline lib/core/monitor_impl.toit]: Should this be moved into acquire_timer_ so you can't forget it?

File: lib/core/monitor_impl.toit [hunk 2]
```diff
@@ -88,6 +88,7 @@ class __Monitor__:
           if deadline and Time.monotonic_us >= deadline: return false
         // Wait until notified. When we get back the monitor might be owned by
         // someone else.
+        if timer: timer.arm deadline
         await_ self
     finally:
       if timer: self.release_timer_ timer
```
Comment (erikcorry) [inline lib/core/monitor_impl.toit:93]: Perhaps move the null check inside release_timer_ ?

**Review by erikcorry — APPROVED** [review]

File: lib/core/monitor_impl.toit (hunk 1 above)
Comment (kasperl) [reply lib/core/monitor_impl.toit]: Done.

File: lib/core/monitor_impl.toit (hunk 2 above)
Comment (kasperl) [reply lib/core/monitor_impl.toit:93]: Keeping as is for consistency with the other 'if timer' checks.

**Pull request merged**